server/src/main/java/org/elasticsearch/repositories/ShardGenerations.java

```diff
@@ -219,6 +221,9 @@ public Builder put(IndexId indexId, int shardId, SnapshotsInProgress.ShardSnapsh
         }
 
         public Builder put(IndexId indexId, int shardId, ShardGeneration generation) {
+            var indexIdWithSameName = indexIdsByName.put(indexId.getName(), indexId);
+            assert indexIdWithSameName == null || indexIdWithSameName.equals(indexId) != false
```

Comment:
`!= false` is redundant :)

```suggestion
            assert indexIdWithSameName == null || indexIdWithSameName.equals(indexId)
```